ddtrace/settings/config.py

```diff
@@ -199,7 +200,8 @@ def __init__(self):
         self.client_ip_header = os.getenv("DD_TRACE_CLIENT_IP_HEADER")
         self.retrieve_client_ip = asbool(os.getenv("DD_TRACE_CLIENT_IP_ENABLED", default=False))
 
-        self.tags = parse_tags_str(os.getenv("DD_TAGS") or "")
+        # cleanup DD_TAGS, because values will be inserted back in the optimal way (via _dd.git.* tags)
+        self.tags = gitmetadata.clean_tags(parse_tags_str(os.getenv("DD_TAGS") or ""))
```

Comment:
I am not sure I like this approach.

If someone is going to set them manually via `DD_TAGS` why should they be removed? That feels like unexpected behavior to just implicitly remove them.

Will this lead to broken scenarios for users? If someone is setting them via `DD_TAGS` today, but is running in an environment where they don't have any of the other env variables or package data set, then we are just going to remove the tags without any warning to the user.

Reply:
1. Tagging can be turned off with an environment variable DD_TRACE_GIT_METADATA_ENABLED
2. If we leave the values here, they will go both through `_dd.git....` and through `git...` which will lead to unnecessary duplication and potential conflicts (_dd.git already propagated to all spans on intake)

Reply:
We already discussed it inside of the team and already implemented (and deployed more than month ago) this logic in Go tracer and already implemented the same for Java tracer. I don't think we need separate logic for Python tracer.

Just for other one clarification, for old users nothing will changed, they can provide all as before via (DD_TAGS) and after intake it will be in the same place (in all spans) and even if they start embed metadata DD_TAGS will be used as preferred way to don't break anything.